cardinal/ecs/component/metadata/component_metadata_test.go

```diff
@@ -27,6 +28,14 @@ type ComponentDataB struct {
 
 func (ComponentDataB) Name() string { return "b" }
 
+func getNameOfComponent(c metadata.Component) string {
+	return c.Name()
+}
+
+func TestComponentInterfaceSignature(t *testing.T) {
+	assert.Equal(t, getNameOfComponent(&ComponentDataA{}), "a")
+}
```

Comment:
does this give us some sort of edge over just directly calling `componentdata.Name()`?

Reply:
No the purpose of this test is to throw a compiler error IF the interface for `metadata.Component` changes. 

So like say one of us decides to change the method to be called String(). This test will throw a compiler error. Actually I should put a comment here to be clear.

Reply:
ahh ok got it

Reply:
actually tbh, this test is a little redundant because a lot of integration tests cover it. But whatever doesn't hurt.